src/EditorFeatures/CSharp/MisplacedUsingDirectives/MisplacedUsingDirectivesInNamespaceDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp;
+
+namespace Microsoft.CodeAnalysis.CSharp.MisplacedUsingDirectives
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class MisplacedUsingDirectivesInNamespaceDiagnosticAnalyzer : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        private static readonly LocalizableResourceString s_localizableOutsideMessage = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Using_directives_must_be_placed_outside_of_a_namespace_declaration), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        public MisplacedUsingDirectivesInNamespaceDiagnosticAnalyzer()
+           : base(IDEDiagnosticIds.MoveMisplacedUsingDirectivesDiagnosticId, MisplacedUsingsUtilities.LocalizableTitle, s_localizableOutsideMessage)
```

Comment:
Interesting pattern: you have two different analyzers reporting the same diagnostic ID and also sharing majority of code through a static utility helper class. Any reason to choose this approach over just having a single analyzer with all the code and registering two separate syntax node actions? You don't need to change the implementation now, but was just curious on the chosen implementation.

Reply:
Note that AbstractCodeStyleDiagnosticAnalyzer does have a .ctor that takes more then one supported descriptors: http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs,57

Reply:
It was suggested I separate the two since they require separate diagnostics because of the different strings. If there is a way to consolidate them cleanly, I would be happy to update.